src/mongocxx/private/index_view.hh

```diff
@@ -134,6 +133,25 @@ class index_view::impl {
             opts_doc.append(bsoncxx::builder::concatenate_doc{session->_get_impl().to_document()});
         }
 
+        if (options.commit_quorum()) {
+            auto server_description = libmongoc::client_select_server(
```

Comment:
`server_description` may be NULL. If so, `error` is set.

If `NULL` is returned, you can use `mongoc_error.hh`'s `throw_exception`, to throw an exception directly from the `bson_error_t`.

Similar to my comment below, perhaps use `write_exception`:

```
throw_exception<write_exception>(error);
```

Secondly, the returned server description needs to be manually destroyed with `mongoc_server_description_destroy`.

Reply:
Done.